chat-client/src/client/chat.ts

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable prefer-const */
+import { TabEventParams } from '@aws/language-server-runtimes/protocol'
+import { CHAT_PROMPT, NEW_TAB_CREATED, ServerMessage, TAB_CHANGED, TAB_REMOVED } from '../contracts/serverContracts'
+import {
+    AUTH_NEEDED_EXCEPTION,
+    ERROR_MESSAGE,
+    SEND_TO_PROMPT,
+    SendToPromptMessage,
+    TAB_ID_RECEIVED,
+    TabIdReceivedParams,
+    UI_FOCUS,
+    UI_IS_READY,
+    UiMessage,
+} from '../contracts/uiContracts'
+import { Messager, OutboundChatApi } from './messager'
+import { InboundChatApi, createMynahUI } from './mynahUi'
+import { TabFactory } from './tabs/tabFactory'
+import { TabStorage } from './tabs/tabStorage'
+
+export const createChat = (clientApi: { postMessage: (msg: UiMessage | ServerMessage) => void }) => {
+    // eslint-disable-next-line semi
+    let mynahApi: InboundChatApi
+
+    const sendMessageToClient = (message: UiMessage | ServerMessage) => {
+        clientApi.postMessage(message)
```

Comment:
What is referred to as `client` here?
If I read this correctly (correcty me if I'm wrong), this is connecting MynahUI Chat Client inside of webview with outside proxy over postMessage in the hosting application. If that's true, can this be renamed to something more explicit, `sendMessageToHost` perhaps?

I found it a bit confusing to understand the direction of messages flow, perhaps changing the name or adding some documentation could help here.

Reply:
`clientApi` is `vscode` api if the extension is `vscode` extension. `postMessage` cannot be renamed to `sendMessageToHost`, `clientApi` can be renamed to whatever makes it more clear.